### PR TITLE
#14 Lays groundwork for maven app to aid gradle addition

### PR DESCRIPTION
**Pull request by willbuck**
So as stated on issue #14, in doing the conversion of the sample app from maven to gradle to support adding gradle as a build option, we found many issues. This resolves those issues, primarily removing some spring dependencies and replacing with micronaut equivalents.

Feedback is welcome!

### CODE REVIEW

File: generators/entity-server/templates/src/main/java/package/common/field_validators.ejs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    if (rules.includes('minlength') && !rules.includes('maxlength')) {
+        validators.push('@Size(min = ' + field.fieldValidateRulesMinlength + ')');
+    }
+    if (rules.includes('maxlength') && !rules.includes('minlength')) {
+        validators.push('@Size(max = ' + field.fieldValidateRulesMaxlength + ')');
+    }
+    if (rules.includes('minlength') && rules.includes('maxlength')) {
+        validators.push('@Size(min = ' + field.fieldValidateRulesMinlength + ', max = ' + field.fieldValidateRulesMaxlength + ')');
+    }
+    // Not supported anymore because the server can't check the size of the blob before downloading it completely.
```
Comment (sendilkumarn) [inline generators/entity-server/templates/src/main/java/package/common/field_validators.ejs]: if this is micronaut blueprint specific todo, then can we follow some standards here?  like ``` TODO: Micronaut <Issue ID> ```

**Comment by sendilkumarn** [comment]
This commit also includes #18 ?

**Comment by JasonTypesCodes** [comment]
> This commit also includes #18 ?

It sure does.  I suspect @willbuck thought #18 would merge first and then this would come afterwards but I suppose if this PR is good it can be used for both.

File: generators/entity-server/templates/src/main/java/package/common/field_validators.ejs (hunk 1 above)
Comment (JasonTypesCodes) [reply generators/entity-server/templates/src/main/java/package/common/field_validators.ejs]: I created #21 to look into this and added a TODO comment to this file referencing it.

**Pull request merged**